ApplensBackend/Middleware/RequestMiddlewareNationalCloud.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace AppLensV3.Middleware
+{
+    public class RequestMiddlewareNationalCloud
+    {
+        private readonly RequestDelegate _next;
+        private readonly string websiteHostName;
+        private readonly string allowedOrigin;
+
+        public RequestMiddlewareNationalCloud(RequestDelegate next)
+        {
+            _next = next;
+            websiteHostName = Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME");
+            allowedOrigin = string.Empty;
+            if (websiteHostName != null)
+            {
+                switch (websiteHostName)
```

Comment:
Instead of switch case on `WEBSITE_HOSTNAME`, lets use the `CloudDomain` of the environment. 

```suggestion
                if (Configuration.IsAzureUSGovernment()) { ... }
```

See [MultiCloudExtensions](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/blob/main/ApplensBackend/Extensions/MultiCloudExtensions.cs)

Reply:
Done